asserts/device_asserts.go

```diff
@@ -0,0 +1,143 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package asserts
+
+import (
+	"fmt"
+	"time"
+)
+
+// Model holds a model assertion, which is a statement by a brand
+// about the properties of a device model.
+type Model struct {
+	assertionBase
+	allowedModes  []string
+	requiredSnaps []string
+	timestamp     time.Time
+}
+
+// BrandID returns the brand identifier. Same as the authority id.
+func (mod *Model) BrandID() string {
+	return mod.Header("brand-id")
+}
+
+// Model returns the model name identifier.
+func (mod *Model) Model() string {
+	return mod.Header("model")
+}
+
+// Series returns the series of the core software the model uses.
+func (mod *Model) Series() string {
+	return mod.Header("series")
+}
+
+// OS returns the OS variant the model uses.
+func (mod *Model) OS() string {
+	return mod.Header("os")
+}
+
+// Architecture returns the archicteture the model is based on.
+func (mod *Model) Architecture() string {
+	return mod.Header("architecture")
+}
+
+// Gadget returns the gadget snap the model uses.
+func (mod *Model) Gadget() string {
+	return mod.Header("gadget")
+}
+
+// Kernel returns the kernel snap the model uses.
+func (mod *Model) Kernel() string {
+	return mod.Header("kernel")
+}
+
+// Store returns the snap store the model uses.
+func (mod *Model) Store() string {
+	return mod.Header("store")
+}
+
+// AllowedModes returns which ones of the "classic" and "developer" modes are allowed for the model.
+func (mod *Model) AllowedModes() []string {
+	return mod.allowedModes
+}
+
+// RequiredSnaps returns the snaps that must be installed at all times and cannot be removed for this model.
+func (mod *Model) RequiredSnaps() []string {
+	return mod.requiredSnaps
+}
+
+// Class returns which class the model belongs to defining policies for
+// additional software installation.
+func (mod *Model) Class() string {
+	return mod.Header("class")
+}
+
+// Timestamp returns the time when the model assertion was issued.
+func (mod *Model) Timestamp() time.Time {
+	return mod.timestamp
+}
+
+// Implement further consistency checks.
+func (mod *Model) checkConsistency(db *Database, acck *AccountKey) error {
+	// TODO: double check trust level of authority depending on class and possibly allowed-modes
+	if !acck.isKeyValidAt(mod.timestamp) {
+		return fmt.Errorf("model assertion timestamp outside of signing key validity")
+	}
+	return nil
+}
+
+var modelMandatory = []string{"os", "architecture", "gadget", "kernel", "store", "class"}
+
+func assembleModel(assert assertionBase) (Assertion, error) {
+	if assert.headers["brand-id"] != assert.headers["authority-id"] {
```

Comment:
@emgee yea I discussed that with Gustavo, the library assumes there is always an authority-id and that's the signer until we get strong input to add the complexity of aliases for it etc